src-java/server42/server42-control-storm-topology/src/main/java/org/openkilda/server42/control/topology/service/FlowRttService.java

```diff
@@ -0,0 +1,94 @@
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.persistence.repositories.FeatureTogglesRepository;
+import org.openkilda.persistence.repositories.FlowRepository;
+import org.openkilda.persistence.repositories.RepositoryFactory;
+import org.openkilda.persistence.repositories.SwitchPropertiesRepository;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class FlowRttService {
+
+    private final IFlowCarrier carrier;
+    private final FeatureTogglesRepository featureTogglesRepository;
+    private final SwitchPropertiesRepository switchPropertiesRepository;
+    private final FlowRepository flowRepository;
+
+    public FlowRttService(IFlowCarrier carrier, PersistenceManager persistenceManager) {
+        this.carrier = carrier;
+        RepositoryFactory repositoryFactory = persistenceManager.getRepositoryFactory();
+        switchPropertiesRepository = repositoryFactory.createSwitchPropertiesRepository();
+        featureTogglesRepository = repositoryFactory.createFeatureTogglesRepository();
+        flowRepository = repositoryFactory.createFlowRepository();
+    }
+
+    /**
+     * Check if rtt global feature enabled and rtt enabled for switch then ask carrier to activate flow.
+     *
+     * @param flowId flow id
+     * @param port switch customer port
+     * @param vlan switch customer vlan id
+     * @param isForward is endpoint forward
+     */
+    public void activateFlowMonitoring(String flowId, SwitchId switchId, Integer port, Integer vlan,
+                                       boolean isForward) {
+        if (isFlowRttFeatureToggle() && isFlowRttFeatureEnabledFor(switchId)) {
+            carrier.notifyActivateFlowMonitoring(flowId, switchId, port, vlan, isForward);
+        } else {
+            log.info("skip activation of flow RTT for flow: {} and switch:{}", flowId, switchId);
+        }
+    }
+
+    /**
+     * Activate monitoring for all existed flows on provided switch.
+     * @param switchId switch id
+     */
+    public void activateFlowMonitoringForSwitch(SwitchId switchId) {
+        Map<Boolean, List<Flow>> flowByDirection =
+                flowRepository.findByEndpointSwitch(switchId).stream().collect(Collectors.partitioningBy(
+                        f -> f.getSrcSwitch().getSwitchId().toLong() == switchId.toLong()));
+
+        flowByDirection.get(true).forEach(flow -> carrier.notifyActivateFlowMonitoring(flow.getFlowId(),
+                switchId, flow.getSrcPort(), flow.getSrcVlan(), true
+        ));
+
+        flowByDirection.get(false).forEach(flow -> carrier.notifyActivateFlowMonitoring(flow.getFlowId(),
+                switchId, flow.getDestPort(), flow.getDestVlan(), false
+        ));
+    }
+
+    private boolean isFlowRttFeatureToggle() {
+        return featureTogglesRepository.find().map(FeatureToggles::getServer42FlowRtt).orElse(false);
```

Comment:
```
        return featureTogglesRepository.find()
                .map(FeatureToggles::getServer42FlowRtt)
                .orElse(FeatureToggles.DEFAULTS.getServer42FlowRtt());
```
Why not?

It will allow us to keep all default values in a single place and change them at any time without worries that some places use hardcoded default values.

Reply:
done